BMSForTesting.h

```diff
@@ -0,0 +1,31 @@
+#include"BMSForCharging.h"
+
+class BatteryMgmtSystmTest : public BatteryManagemtSystem
+{
+public:
+    BatteryMgmtSystmTest(int temperature, int soc, float chargeRate):BatteryManagemtSystem(temperature,soc,chargeRate)
+    {}
+    bool monitorTemparature(){
+        if(getTemperature() < TEMPERATURE_LOWER_LIMIT || getTemperature() > TEMPERATURE_UPPER_LIMIT) {
+            cout << "Temperature out of range!\n";
+            return false;
+        }
+        return true;
+    }
+    bool monitorSoc(){
+        if(getSoc() < SOC_LOWER_LIMIT || getSoc() > SOC_UPPER_LIMIT) {

```

Comment:
Similar range checking duplicate code can be reduced and refactored to a single method. for ex. IsValueInRange(...)